spec/lib/ecco/row_event_listener_spec.rb

```diff
@@ -53,6 +56,17 @@
 
         context "UPDATE_ROWS" do
           let(:row_event_type) { EventType::UPDATE_ROWS }
+          let(:rows) do
+            ArrayList.new(
+              Arrays.asList(
+                AbstractMap::SimpleEntry.new(
+                  Array.new(2).to_java,
+                  Array.new(2).to_java
+                )
+              )
+            )
+          end
+
```

Comment:
I tried the following to get rid of the `let(:rows)` business from this file. Just to make sure all rows are defined in the same file (`event_context.rb`). The imports needs to move too.

WDYT?

This could perhaps be cleaned up a bit, but I have no good ideas right now.

```diff
diff --git a/spec/event_context.rb b/spec/event_context.rb
index d24c178..a303d81 100644
--- a/spec/event_context.rb
+++ b/spec/event_context.rb
@@ -3,13 +3,36 @@ java_import java.util.Arrays
 
 shared_context "event" do
   let(:table_id) { 1 }
-  let(:rows) do
+
+  def update_event?
+    [
+      EventType::EXT_UPDATE_ROWS,
+      EventType::UPDATE_ROWS,
+    ].include?(row_event_type)
+  end
+
+  let(:rows_for_update_event) do
+    ArrayList.new(
+      Arrays.asList(
+        AbstractMap::SimpleEntry.new(
+          Array.new(2).to_java,
+          Array.new(2).to_java
+        )
+      )
+    )
+  end
+
+  let(:rows_for_write_and_delete_events) do
     Java::JavaUtil::LinkedList.new(
       Arrays.asList(
         Array.new(1).to_java { Array.new(2).to_java }
       )
     )
   end
+
+  let(:rows) do
+    update_event? ? rows_for_update_event : rows_for_write_and_delete_events
+  end
   let(:database) { "some_database" }
   let(:table) { "some_table" }
   let(:table_event_type) { EventType::TABLE_MAP }
diff --git a/spec/lib/ecco/row_event_listener_spec.rb b/spec/lib/ecco/row_event_listener_spec.rb
index a476845..e051b2c 100644
--- a/spec/lib/ecco/row_event_listener_spec.rb
+++ b/spec/lib/ecco/row_event_listener_spec.rb
@@ -56,17 +56,6 @@ describe Ecco::RowEventListener do
 
         context "UPDATE_ROWS" do
           let(:row_event_type) { EventType::UPDATE_ROWS }
-          let(:rows) do
-            ArrayList.new(
-              Arrays.asList(
-                AbstractMap::SimpleEntry.new(
-                  Array.new(2).to_java,
-                  Array.new(2).to_java
-                )
-              )
-            )
-          end
-
           it "should return UPDATE_ROWS" do
             actual_type = nil
             subject.callback = Proc.new do |row_event|
@@ -111,17 +100,6 @@ describe Ecco::RowEventListener do
 
         context "EXT_UPDATE_ROWS" do
           let(:row_event_type) { EventType::EXT_UPDATE_ROWS }
-          let(:rows) do
-            ArrayList.new(
-              Arrays.asList(
-                AbstractMap::SimpleEntry.new(
-                  Array.new(2).to_java,
-                  Array.new(2).to_java
-                )
-              )
-            )
-          end
-
           it "should return UPDATE_ROWS" do
             actual_type = nil
             subject.callback = Proc.new do |row_event|
```

Reply:
Oh nice! This is very helpful, gonna take a closer look soon. :)

Reply:
I don't know how to do this cleaner either, gonna check a little more. Overall I think this change is good, makes the spec file a bit less cluttered.

Reply:
```diff
diff --git a/spec/event_context.rb b/spec/event_context.rb
index d24c178..e0623e7 100644
--- a/spec/event_context.rb
+++ b/spec/event_context.rb
@@ -3,17 +3,39 @@ java_import java.util.Arrays
 
 shared_context "event" do
   let(:table_id) { 1 }
+  let(:database) { "some_database" }
+  let(:table) { "some_table" }
+  let(:table_event_type) { EventType::TABLE_MAP }
+  let(:row_event_type) { EventType::EXT_WRITE_ROWS }
   let(:rows) do
+    update_event? ? rows_for_update_event : rows_for_write_and_delete_events
+  end
+
+  def update_event?
+    [
+      EventType::EXT_UPDATE_ROWS,
+      EventType::UPDATE_ROWS,
+    ].include?(row_event_type)
+  end
+
+  let(:rows_for_update_event) do
+    ArrayList.new(
+      Arrays.asList(
+        AbstractMap::SimpleEntry.new(
+          Array.new(2).to_java,
+          Array.new(2).to_java
+        )
+      )
+    )
+  end
+
+  let(:rows_for_write_and_delete_events) do
     Java::JavaUtil::LinkedList.new(
       Arrays.asList(
         Array.new(1).to_java { Array.new(2).to_java }
       )
     )
   end
-  let(:database) { "some_database" }
-  let(:table) { "some_table" }
-  let(:table_event_type) { EventType::TABLE_MAP }
-  let(:row_event_type) { EventType::EXT_WRITE_ROWS }
 
   let(:table_event) do
     event = double("Event")

```

Reply:
I rearranged the order a little bit, maybe this is better? :)

Reply:
:+1: Yes, I think that looks better having the `let`s grouped that way :)

Reply:
Bam 436955b6e3ee90c599ec90c9176f2d6d82b05c0e